File: sorting_algorithm/bubble_sort/js/bubble_sort.js
function bubbleSort(items) {
  var itemsLength = items.length;

  for (var i = 0; i < itemsLength; i++) {
    for (var j = 0; j < itemsLength - 1 - i; j++) {
      if (items[j] > items[j + 1]) {
        var temp = items[j];
        items[j] = items[j + 1];
        items[j + 1] = temp;
      }
    }
  }

  return items;
}

(function() {
  var items = [1, 95, 66, 4, 7, 21, 9, 5, 14, 86, 94, 11];
  var sorted_items = bubbleSort(items);
  
  console.log(sorted_items);
})();

File: sorting_algorithm/bubble_sort/dart/bubble_sort.dart
List<dynamic> bubbleSort(List<dynamic> list) {
  int listSize = list.length;

  for (int i = 0; i < listSize; i++) {
    for (int j = 0; j < listSize - 1 - i; j++) {
      if (list[j] > list[j + 1]) {
        var temp = list[j];
        list[j + 1] = list[j];
        list[j] = temp;
      }
    }
  }

  return list;
}

void main() {
  List<dynamic> list = [1, 4, 24, 8, 33, 14, 2, 9, 17, 95];
  print(bubbleSort(list));
}
